oauth2client/service_account.py

```diff
@@ -461,3 +465,96 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+def _datetime_to_secs(utc_time):
+   # time_delta.total_seconds() not supported in Python 2.6
+   epoch = datetime.datetime(1970, 1, 1)
+   time_delta = utc_time - epoch
+   return time_delta.days * 86400 + time_delta.seconds
+
+
+class JWTAccessCredentials(OAuth2Credentials):
+    """Self signed JWT credentials.
+
+    Makes an assertion to server using a self signed JWT from service account
+    credentials.  These credentials do NOT use OAuth 2.0 and instead authenticate
+    directly.
+    """
+    MAX_TOKEN_LIFETIME_SECS = 3600
+    """Max lifetime of the token (one hour, in seconds)."""
+
+    def __init__(self, service_account_creds, additional_claims=None, access_token=None,
+                 token_expiry=None):
+        super(JWTAccessCredentials, self).__init__(access_token, None, None, None, token_expiry,
+                                                   None, None)
+        self._creds = service_account_creds
+        self._claims = {}
+        if additional_claims:
+            self._claims.update(additional_claims)
+    
+    def get_access_token(self, http=None, additional_claims=None):
+        """Create a signed jwt
+        Agrs:
+            additional_claims: dict, additional claims to add 
+                the payload of the JWT. 
+            Returns:
+                An AccessTokenInfo with the signed jwt
+        """
+        if additional_claims is not None:
+            # Create a 1 time token
+            token, expiry = self._create_token(additional_claims)
+            return token
+        else:
+            if self.access_token is None or self.access_token_expired:
+                self._refresh(None)
+
+        return AccessTokenInfo(access_token=self.access_token,
+                              expires_in=self._expires_in())
+    def revoke(self, http):
+        raise TokenRevokeError('Cannot revoke JWTAccessCredentials')
+
+    @classmethod
+    def from_json(cls, json_data):
+        json_data = json.loads(_from_bytes(json_data))
+
+        service_account_creds = ServiceAccountCredentials.from_json(json_data['_creds'])
+        token_expiry = datetime.datetime.strptime(json_data['token_expiry'], EXPIRY_FORMAT)
+        retval = cls(service_account_creds, 
+                     additional_claims=json_data['_claims'],
+                     access_token=json_data['access_token'],
```

Comment:
Since the access token is an assertion, is it necessary to load this from json? Can't it just be re-generated?